sqlalchemy_access/base.py

```diff
@@ -13,11 +13,20 @@
 """
 from sqlalchemy import sql, schema, types, exc, pool
 from sqlalchemy.sql import compiler, expression
+from sqlalchemy.sql.compiler import OPERATORS
+from sqlalchemy.sql.operators import ne, is_, isnot
 from sqlalchemy.engine import default, base, reflection
 from sqlalchemy import processors
 
 import pyodbc
 
+# These operator overrides are required in order to make
+# the AccessCompiler class correctly format emmited queries
+OPERATORS[ne] =  " <> "
+OPERATORS[is_] =  " = "
```

Comment:
I wasn't aware of that in particular, but in my testing it appears that Access is finicky about IS comparisons to True and False, but isn't finicky about the same comparisons using = .

The behavior on the Access side of things appears to be identical if you use = for comparisons to True, False, and NULL. Access throws an error when I try to use IS for anything *but* NULL. Given that, it comes down to a code style thing I think. It seems more Pythonic to my eye to use `.is_(True)` rather than ` == True `

Forgoing the override causes SQLAlchemy to throw an exception when trying to do things like `session.query(Sale).filter(Sale.taxable.is_(True)).all()`, which forces you to use `session.query(Sale).filter(Sale.taxable == True).all()` or `session.query(Sale).filter(Sale.taxable == 1).all()` (both of which appear to work, but just aren't quite as Pythic feeling.)

Making the change still allows the use of ` == ` on the Python side of things, it just *also* allows the use of ` .is_() ` and ` .isnot_() ` to function as expected. With no negative or restrictive impact on usability, it seemed silly not to include it.

Reply:
Access "Yes/No" columns are a bit of a special case because they are not nullable. If we try to assign a `NULL` value to that column it is written to the database as `False`. Thus, `WHERE colname = NULL` and `WHERE colname IS NULL` will both always return zero rows. However, for a nullable column type like "Short Text" `WHERE colname IS NULL` will return the rows where the column really is NULL, while `WHERE colname = NULL` will always return zero rows.

TL;DR: `is` and equality really do mean different things, both in SQL and [in Python](https://stackoverflow.com/q/132988/2144390).

Reply:
Ah. Well, thank you for clearing that up for me. I'll edit the patch.